src/App.tsx

```diff
@@ -1,17 +1,128 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 
 import users from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+
+import { User, PrepTodo } from './react-app-env';
+
+const getUserById = (userId: number): User | null => {
+  return users.find(user => user.id === userId) || null;
+};
+
+const todosWithUser: PrepTodo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 const App: React.FC = () => {
+  const [todos, setTodos] = useState<PrepTodo[]>(todosWithUser);
+  const [newTodoName, setNewTodoName] = useState('');
+  const [hasNameError, setHasNameError] = useState(false);
+
+  const [selectUserId, setSelectUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const addTodo = (name: string, userId: number) => {
+    const newTodo: PrepTodo = {
+      id: Date.now(),
+      name,
+      userId,
+      completed: false,
+      user: getUserById(selectUserId),
+    };
+
+    setTodos((prevTodos) => [...prevTodos, newTodo]);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setHasNameError(!newTodoName);
+    setHasUserIdError(!selectUserId);
+
+    if (newTodoName && selectUserId) {
+      addTodo(newTodoName, selectUserId);
+      setNewTodoName('');
+      setSelectUserId(0);
+    }
+  };
+
+  // const todosWithUser: PrepTodo[] = todos.map(todo => ({
+  //   ...todo,
+  //   user: getUserById(todo.userId),
+  // }));
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
+      <form
+        onSubmit={handleSubmit}
+      >
+        <label className="">
+          <p className="App__title">Title</p>
+          <input
+            type="text"
+            value={newTodoName}
+            placeholder="write to do"
+            onChange={(event) => {
```

Comment:
It'll be better to transfer callback in onChange event before return and pass only function link.